lib/galaxy/managers/datasets.py

```diff
@@ -365,6 +314,85 @@ def extra_files(self, dataset_assoc):
             return []
         return glob.glob(os.path.join(dataset_assoc.dataset.extra_files_path, '*'))
 
+    def serialize_dataset_association_roles(self, trans, dataset_assoc):
+        if hasattr(dataset_assoc, "library_dataset_dataset_association"):
+            library_dataset = dataset_assoc
+            dataset = library_dataset.library_dataset_dataset_association.dataset
+        else:
+            library_dataset = None
+            dataset = dataset_assoc.dataset
+
+        # Omit duplicated roles by converting to set
+        access_roles = set(dataset.get_access_roles(trans))
+        manage_roles = set(dataset.get_manage_permissions_roles(trans))
+
+        access_dataset_role_list = [(access_role.name, trans.security.encode_id(access_role.id)) for access_role in access_roles]
+        manage_dataset_role_list = [(manage_role.name, trans.security.encode_id(manage_role.id)) for manage_role in manage_roles]
+        rval = dict(access_dataset_roles=access_dataset_role_list, manage_dataset_roles=manage_dataset_role_list)
+        if library_dataset is not None:
+            modify_roles = set(trans.app.security_agent.get_roles_for_action(library_dataset, trans.app.security_agent.permitted_actions.LIBRARY_MODIFY))
+            modify_item_role_list = [(modify_role.name, trans.security.encode_id(modify_role.id)) for modify_role in modify_roles]
+            rval["modify_item_roles"] = modify_item_role_list
+        return rval
+
+    def update_permissions(self, trans, dataset_assoc, **kwd):
+        action = kwd.get('action', 'set_permissions')
+        if action not in ['remove_restrictions', 'make_private', 'set_permissions']:
```

Comment:
What was your feeling re: `make_public` (or `make_public_by_link` instead of `remove_restrictions`?)?

Reply:
I thought we could add that as an alias but I wanted to maintain the previous action name also for backward compatibility in data libraries. I'll make this change!

Reply:
Backwards compat is a great reason! If that's what devs/clients are expecting to need, definitely no reason to add something else we'll accept.

Reply:
Fair, thanks!